### PR TITLE
Fix library to work with API changes after 17 Nov 2015

### PR DESCRIPTION
**Pull request by florianbeer**
- Add new permission scopes
- Require access token for all endpoints
- Remove deprecated endpoints
- Fix parameters for endpoints
- Fix pagination

### CODE REVIEW

**Comment by fabiomlferreira** [comment]
Please approve this request

**Comment by filkaris** [comment]
I agree, I was about to make a PR to add new permission scopes and I saw this.
It looks ok and it's important, please merge this.

File: src/Instagram.php [hunk 1]
```diff
@@ -525,16 +548,14 @@ public function pagination($obj, $limit = 0)
...
-            $auth = (strpos($apiCall[1], 'access_token') !== false);
-
-            if (isset($obj->pagination->next_max_id)) {
-                return $this->_makeCall($function, $auth, array('max_id' => $obj->pagination->next_max_id, 'count' => $limit));
+            if (isset($obj->pagination->next_max_tag_id)) {
+                return $this->_makeCall($function, array('max_tag_id' => $obj->pagination->next_max_tag_id, 'count' => $count));
             }
 
-            return $this->_makeCall($function, $auth, array('cursor' => $obj->pagination->next_cursor, 'count' => $limit));
+            return $this->_makeCall($function, array('next_max_id' => $obj->pagination->next_max_id, 'count' => $count));
```
Comment (dgsjm) [inline src/Instagram.php:597]: The pagination call is made with a parameter named "next_max_id". It seems that it should be named "max_id". For example, see the parameters for the ["/users/self/media/recent/" endpoint](https://www.instagram.com/developer/endpoints/users/#get_users_media_recent_self). Am I missing something?

Comment (florianbeer) [inline src/Instagram.php:597]: Pagination seems to work for me, but maybe I didn't use that endpoint yet.
Did you test it out? What is the error message you are getting back from the Twitter API?
If this is indeed an error, please make a pull request with the changed parameter name.
Thank you.

Comment (dgsjm) [inline src/Instagram.php:597]: I don't get any error from Instagram, but each pagination call returns the same batch of results. Changing "next_max_id" to "max_id" allowed me to get the next batch of items in the paginated list.

**Comment by angelxmoreno** [comment]
:shipit:

**Pull request closed without merging**